tests/unit/conftest.py

```diff
@@ -134,3 +142,45 @@ def validate_target_transpile(input_sql, *, target=None, pretty=False):
 @pytest.fixture(scope="session")
 def dialect_context():
     yield validate_source_transpile, validate_target_transpile
+
+
+def parse_sql_files(input_dir: Path, source: str, target: str, is_expected_exception):
+    suite = []
+    for dirpath, _, filenames in os.walk(input_dir):
+        if dirpath.endswith("__pycache__"):
+            continue
+        for file in filenames:
+            if not file.endswith('.sql'):
+                continue
+            abs_path = Path(dirpath) / file
+            with open(abs_path, 'r', encoding="utf-8") as file_content:
+                content = file_content.read()
+            if content:
+                parts = content.split(f"-- {source.lower()} sql:")
+                for part in parts[1:]:
+                    source_sql = re.split(r'-- \w+ sql:', part)[0].strip().rstrip(";")
+                    target_sql = (
+                        re.split(rf'-- {target} sql:', part)[1]
+                        if len(re.split(rf'-- {target} sql:', part)) > 1
+                        else re.split(r'-- databricks sql:', part)[1]
+                    )
+                    target_sql = re.split(r'-- \w+ sql:', target_sql)[0].strip().rstrip(';').replace('\\', '')
+                    # when multiple sqls are present below target
+                    test_name = file.replace('.sql', '')
+                    if is_expected_exception:
+                        suite.append(
+                            FunctionalTestFileWithExpectedException(
+                                target_sql, source_sql, test_name, expected_exceptions[test_name]
+                            )
+                        )
+                    else:
+                        suite.append(FunctionalTestFile(target_sql, source_sql, test_name))
+    return suite
+
+
+def get_functional_test_files_from_directory(
+    input_dir: Path | str, source: str, target: str, is_expected_exception=False
```

Comment:
```suggestion
    input_dir: Path, source: str, target: str, is_expected_exception=False
```
unions of incompatible types are to be avoided.

Reply:
Done